Packs/Netskope/Integrations/NetskopeEventCollector_v2/NetskopeEventCollector_v2.py

```diff
@@ -0,0 +1,651 @@
+
+            if is_re_fetch_failed_fetch:
+                # in case of re-fetch failures we don't use pagination, just the fetch the failed chunk
+                total_events = limit
+                max_offset = init_offset + int(limit)
+            else:
+                total_events = await client.get_events_count(type, request_params)
+                max_offset = min(total_events, init_offset + int(limit))
+
+            request_limit = request_params.get("limit", MAX_EVENTS_PAGE_SIZE)
+            demisto.debug(
+                f"Going to fetch {min(total_events, limit)} events from {init_offset=} by chunks of {request_limit} ..."
+            )
+            tasks = [
+                # asyncio.create_task(_handle_page(request_params | {'offset': offset}))
+                _handle_page(request_params | {"offset": offset})
+                for offset in range(init_offset, max_offset, request_limit)
+            ]
+            results = await asyncio.gather(*tasks, return_exceptions=True)
+            return results
+        except Exception as e:
+            raise DemistoException(message=str(e), exception=e, res=request_params)
+
+    try:
+        results: list[list[dict] | BaseException] = await _handle_all_pages()
+        success_tasks = list(filter(lambda res: not isinstance(res, BaseException), results))
+        failures = list(filter(lambda res: isinstance(res, BaseException), results))
+        return success_tasks, failures
+    except Exception as e:
+        return [], [e]
+
+
+""" COMMANDS FUNCTIONS """
+
+
+async def handle_fetch_and_send_all_events(
+    client: Client, last_run: dict, limit: int = MAX_EVENTS_PAGE_SIZE, send_to_xsiam=False
+) -> tuple[list[dict], dict]:
+    """
+    Iterates over all supported event types and call the handle event fetch logic and send the events to XSIAM.
+
+    Endpoint: /api/v2/events/data/
+    Docs: https://www.postman.com/netskope-tech-alliances/netskope-rest-api/request/zknja6y/get-network-events-generated-by-netskope
+
+    Example HTTP request:
+    <baseUrl>/api/v2/events/data/network?offset=0&insertionstarttime=1707466628&insertionendtime=1739089028&
+
+    Args:
+        client (Client): The Netskope client.
+        last_run (dict): The execution last run dict where the relevant operations are stored.
+        limit (int): The limit which after we stop pulling.
+        send_to_xsiam(bool): Whether to send the fetched events to XSIAM or not.
+
+    Returns:
+        list: The accumulated list of all events.
+        dict: The updated last_run object.
+    """
+    start = time.time()
+    # needed as we use concurrent async tasks
+    support_multithreading()
+
+    remove_unsupported_event_types(last_run, client.event_types_to_fetch)
+
+    all_events = []
+    epoch_current_time = str(int(arg_to_datetime("now").timestamp()))  # type: ignore[union-attr]
+    epoch_last_month = str(int(arg_to_datetime("1 day").timestamp()))  # type: ignore[union-attr]
```

Comment:
```suggestion
    epoch_last_day = str(int(arg_to_datetime("1 day").timestamp()))  # type: ignore[union-attr]
```
Consider fixing the type so the ignore is not neeeded.